src/build/cargo.rs

```diff
@@ -367,25 +354,22 @@ impl Executor for RlsExecutor {
             // so the dep-info is ready by the time we return from this callback.
             // NB: In `workspace_mode` regular compilation is performed here (and we don't
             // only calculate dep-info) so it should fix the problem mentioned above.
-            // Since ProcessBuilder doesn't allow to modify args, we need to create
-            // our own command here from scratch here.
-            for a in &args {
-                // Emitting only dep-info is possible only for final crate type, as
-                // as others may emit required metadata for dependent crate types
-                if a.starts_with("--emit") && is_final_crate_type && !self.workspace_mode {
-                    cmd.arg("--emit=dep-info");
-                } else if a != "-Zsave-analysis" {
```

Comment:
Right, that's for the primary crate units, my bad. But does that mean that when compiling lib for the final bin target, we don't generate save-analysis as well? Shouldn't we? Or am I missing something still?